rdtools/system_analysis.py

```diff
@@ -0,0 +1,585 @@
+                temperature = normalization.interpolate(temperature, interp_freq, max_timedelta)
+            if clearsky_poa is not None:
+                clearsky_poa = normalization.interpolate(clearsky_poa, interp_freq, max_timedelta)
+            if clearsky_temperature is not None:
+                clearsky_temperature = normalization.interpolate(clearsky_temperature, interp_freq, max_timedelta)
+            if isinstance(pv_azimuth, (pd.Series, pd.DataFrame)):
+                pv_azimuth = normalization.interpolate(pv_azimuth, interp_freq, max_timedelta)
+            if isinstance(pv_tilt, (pd.Series, pd.DataFrame)):
+                pv_tilt = normalization.interpolate(pv_tilt, interp_freq, max_timedelta)
+
+        if pv_input == 'power':
+            self.pv_power = pv
+            self.pv_energy = normalization.energy_from_power(pv, max_timedelta=max_timedelta)
+        elif pv_input == 'energy':
+            self.pv_power = None
+            self.pv_energy = pv
+
+        if temperature_input == 'cell':
+            self.cell_temperature = temperature
+            self.ambient_temperature = None
+        elif temperature_input == 'ambient':
+            self.cell_temperature = None
+            self.ambient_temperature = temperature
+
+        if clearsky_temperature_input == 'cell':
+            self.clearsky_cell_temperature = clearsky_temperature
+            self.clearsky_ambient_temperature = None
+        elif clearsky_temperature_input == 'ambient':
+            self.clearsky_cell_temperature = None
+            self.clearsky_ambient_temperature = clearsky_temperature
+
+        self.poa = poa
+        self.temperature_coefficient = temperature_coefficient
+        self.aggregation_freq = aggregation_freq
+        self.pvlib_location = pvlib_location
+        self.clearsky_poa = clearsky_poa
+        self.windspeed = windspeed
+        self.albedo = albedo
+        self.temperature_model = temperature_model
+        self.pv_azimuth = pv_azimuth
+        self.pv_tilt = pv_tilt
+        self.pv_nameplate = pv_nameplate
+        self.results = {}
+
+        # Initialize to use default filter parameters
+        self.filter_params = {
+            'normalized_filter': {},
+            'poa_filter': {},
+            'tcell_filter': {},
+            'clip_filter': {},
+            'csi_filter': {},
+            'ad_hoc_filter': None  # use this to include an explict filter
+        }
+
+    def calc_clearsky_poa(self, times=None, rescale=True, model='isotropic', **kwargs):
+        '''
+        Calculate clearsky plane-of-array irradiance and store in self.clearsky_poa
+
+        Parameters
+        ----------
+        times: Pandas DateTimeIndex for which to calculate clearsky poa
+        rescale: Whether to attempt to rescale clearsky irradiance to measured (bool)
+        model: Model for pvlib.irradiance.get_total_irradiance() (str)
+        kwargs: key word arguments passed to pvlib.irradiance.get_total_irradiance()
```

Comment:
I'd lean toward including the entire function name e.g. `get_total_irradiance_kwargs`. But is that too verbose?

There are a number of places we pass kwargs to lower level functions. If we make this change we should change them all.

Reply:
Since the function signature uses the double-splat operator, the user can't directly specify a value for `kwargs`, they can only specify keyword arguments that will then get collected into `kwargs`.  I.e. code like this isn't valid: `calc_clearsky_poa(self, times=..., rescale=..., model='isotropic', kwargs={'key1': 'value1'})`.  So from the user's perspective, I think the only place they would ever see the name of that variable is if they were looking at the docstring, in which case it's pretty clear what it's used for.  I'd say leave the name as-is since `**kwargs` is idiomatic python.  

Also pvlib uses `kwargs` the way this PR currently does, to pass on to the lower-level function: https://pvlib-python.readthedocs.io/en/stable/generated/pvlib.pvsystem.PVSystem.get_irradiance.html#pvlib.pvsystem.PVSystem.get_irradiance

I do have some small preference for pvlib's wording though -- `Extra parameters passed to xyz` vs `key word arguments passed to xyz`.

Reply:
Thanks @kevinsa5, I think that makes sense.